azurerm/internal/services/apimanagement/api_management_api_resource.go

```diff
@@ -306,6 +353,24 @@ func resourceArmApiManagementApiCreateUpdate(d *schema.ResourceData, meta interf
 	subscriptionKeyParameterNamesRaw := d.Get("subscription_key_parameter_names").([]interface{})
 	subscriptionKeyParameterNames := expandApiManagementApiSubscriptionKeyParamNames(subscriptionKeyParameterNamesRaw)
 
+	authenticationSettings := &apimanagement.AuthenticationSettingsContract{}
+	if vs, hasOAuth2Authorization := d.GetOk("oauth2_authorization"); hasOAuth2Authorization {
+		oAuth2AuthorizationVs := vs.([]interface{})
+		oAuth2AuthorizationV := oAuth2AuthorizationVs[0].(map[string]interface{})
```

Comment:
Can we wrap these in expand functions to guard against crashes?

Reply:
Done in 778a38832e0127eaaf2fb5885eabc16ba90a3405.